File: Npc/11003270.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11003270: Jin
/// </summary>
public class _11003270 : NpcScript {
    protected override int First() {
        return 30;
    }

    // Select 0:
    // $script:0403155707008222$
    // - What brings you here?
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (30, 0):
                // $script:0403155707008223$
                // - If only Captain Stilton were still alive... 
                return -1;
        }
        
        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (30, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11003270.py
""" 11003270: Jin """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0403155707008222$
        # - What brings you here?
        return None # TODO

    def __30(self, pick: int) -> int:
        # $script:0403155707008223$
        # - If only Captain Stilton were still alive... 
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.CLOSE
        return Option.NONE
